pip/utils/outdated.py

```diff
@@ -121,18 +121,19 @@ def pip_version_check(session):
 
         # Refresh the version if we need to or just see if we need to warn
         if pypi_version is None:
-            resp = session.get(
-                PyPI.pip_json_url,
-                headers={"Accept": "application/json"},
+            # Lets use PackageFinder to see what the latest pip version is
+            finder = PackageFinder(
+                find_links=options.find_links,
+                index_urls=[options.index_url] + options.extra_index_urls,
+                allow_all_prereleases=False,  # Explicitly set to False
+                trusted_hosts=options.trusted_hosts,
+                process_dependency_links=options.process_dependency_links,
+                session=session,
             )
-            resp.raise_for_status()
-            pypi_version = [
-                v for v in sorted(
-                    list(resp.json()["releases"]),
-                    key=packaging_version.parse,
-                )
-                if not packaging_version.parse(v).is_prerelease
-            ][-1]
+            # Obtain the newest pacakge from the InstallationCandidate list
+            # When sorted it will be the last element
+            all_candidates = sorted(finder.find_all_candidates("pip"))
```

Comment:
No need to sort the list, getting the max:

    all_candidates = finder.find_all_candidates("pip")
    if not all_candidates:
        return
    pypi_version = max(all_candidates, key=lambda c: c.version).version

should be enough.